integration/rawtx_test.go

```
@@ -0,0 +1,203 @@
+	"encoding/hex"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/integration/rpctest"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/stretchr/testify/require"
+)
+
+// TestTestMempoolAccept checks that `TestTestMempoolAccept` behaves as
+// expected. It checks that,
+// - an error is returned when invalid params are used.
+// - orphan tx is rejected.
+// - fee rate above the max is rejected.
+// - a mixed of both allowed and rejected can be returned in the same response.
+func TestTestMempoolAccept(t *testing.T) {
+	t.Parallel()
+
+	// Boilerplate codetestDir to make a pruned node.
+	btcdCfg := []string{"--rejectnonstd", "--debuglevel=debug"}
+	r, err := rpctest.New(&chaincfg.SimNetParams, nil, btcdCfg, "")
+	require.NoError(t, err)
+
+	// Setup the node.
+	require.NoError(t, r.SetUp(true, 100))
+	t.Cleanup(func() {
+		require.NoError(t, r.TearDown())
+	})
+
+	// Create testing txns.
+	invalidTx := decodeHex(t, missingParentsHex)
+	validTx := createTestTx(t, r)
+
+	// Create testing constants.
+	const feeRate = 10
+
+	testCases := []struct {
+		name           string
+		txns           []*wire.MsgTx
+		maxFeeRate     float64
+		expectedErr    error
+		expectedResult []*btcjson.TestMempoolAcceptResult
+	}{
+		{
+			// When too many txns are provided, the method should
+			// return an error.
+			name:           "too many txns",
+			txns:           make([]*wire.MsgTx, 26),
+			maxFeeRate:     0,
+			expectedErr:    rpcclient.ErrInvalidParam,
+			expectedResult: nil,
+		},
+		{
+			// When no txns are provided, the method should return
+			// an error.
+			name:           "empty txns",
+			txns:           nil,
+			maxFeeRate:     0,
+			expectedErr:    rpcclient.ErrInvalidParam,
+			expectedResult: nil,
+		},
+		{
+			// When a corrupted txn is provided, the method should
+			// an error.
```

nit:  "should return an error"

---

fixed!